pkg/util/limiter/query_limiter.go

```diff
@@ -15,25 +16,32 @@ import (
 type queryLimiterCtxKey struct{}
 
 var (
-	ctxKey          = &queryLimiterCtxKey{}
-	errMaxSeriesHit = "The query hit the max number of series limit (limit: %d)"
+	ctxKey              = &queryLimiterCtxKey{}
+	errMaxSeriesHit     = "The query hit the max number of series limit (limit: %d)"
+	errMaxChunkBytesHit = "The query hit the max number of chunk bytes limit (limit: %d)"
 )
 
 type QueryLimiter struct {
 	uniqueSeriesMx sync.Mutex
 	uniqueSeries   map[model.Fingerprint]struct{}
 
-	maxSeriesPerQuery int
+	chunkBytesCount *atomic.Int32
+
+	maxSeriesPerQuery     int
+	maxChunkBytesPerQuery int
```

Comment:
This limits us to 2GB (2^31 -1 bytes) per query, is it worth making this an unsigned int which is about 4GB (2^32 bytes) per query or a 64 bit number?

Reply:
On 64-bit systems, `int` is 64-bit, so this is fine. Note that Cortex officially doesn't support 32-bit systems.

Reply:
I would be explicit like we do everywhere else.

Reply:
Should we also pass in an int64 at the config/limit.go level? Or is leaving NewQueryLimiter(int, int) and casting the maxChunkBytes value to an int64 ok?

Reply:
> I would be explicit like we do everywhere else.

I don't think we're explicit "everywhere else". I think it would make sense to use `int` here simply because we cannot fit more than `max of int` into memory anyway (applies for both 32-bit and 64-bit platforms).

Reply:
To your question Tyler, if you go with `int64` route, you will need to "extend" that everywhere to avoid losing precision somewhere (ie. in `NewQueryLimiter` too)

Reply:
Ok. Let's not block on this and keep `int`.